### PR TITLE
7pm-3 md - Add swagger-ui/index.html and api/docs endpoints

### LINKED ISSUE #257: Add Swagger UI to assist in developing backend routes
# User Story

As a developer, I can use `swagger-ui` to access backend endpoints easily, so that it is easier to do backend development.

# Discussion

You may use this PR from `proj-ucsb-courses-search` as a roadmap to know what to do: ucsb-cs156-s21/proj-ucsb-courses-search#224

# Acceptance Criteria

- [x] There is a URL that can be accessed e.g. /swagger-ui/index.html to bring up the backend swagger web page
- [x] We can interact with backend APIs through that web page.

### PR DESCRIPTION
**Pull request by mdeirossi**
In this PR, we add the `swagger-ui/index.html` endpoint for API documentation and testing with a UI, and we add the JSON endpoint `api/docs`, following the implementation described in the [courses-search pull request](https://github.com/ucsb-cs156-s21/proj-ucsb-courses-search/pull/224).

Authenticated endpoints require a token. One can be obtained if and only if you are already logged in, by copying from the developer tools after making a request within the app, then pasting into the Authentication field on the swagger-ui page.

Annotations for additional documentation of endpoints can be added in the future.
Closes #257

### CODE REVIEW

File: src/main/java/edu/ucsb/mapache/config/SpringFoxConfig.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package edu.ucsb.mapache.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+// import springfox.documentation.builders.PathSelectors;
```
Comment (andrewhlu) [inline src/main/java/edu/ucsb/mapache/config/SpringFoxConfig.java:6]: Do we need this line? If not, we should remove it.

**Review by andrewhlu — CHANGES_REQUESTED** [review]
see above

**Review by TranNicholas — APPROVED** [review]
LGTM

**Review by andrewhlu — APPROVED** [review]
Tested on Mapache Search QA, looks good!
Thanks for getting this in so quickly, this will surely help everyone else with their code 😄

**Pull request merged**